### PR TITLE
Added max rewards to RngRelayAuction

### PR DESCRIPTION
**Pull request by asselstine**

### CODE REVIEW

**Comment by trmid** [comment]
We didn't get a chance yet to discuss this topic so here are my thoughts:

If we are setting a hard-cap to the max reward, I think it's best if we have a way for governance to update it if the environment changes drastically.

If we want to avoid manual controls, we could also explore setting a dynamic function that lets the max reward fluctuate as a function of the last auction sales.

**Comment by PierrickGT** [comment]
I'm not even sure we should cap it.
If the reserve accrue a lot of POOL, it would mean that people didn't think it was worth awarding the pool in the first place no?
The market should take care of it.

**Comment by asselstine** [comment]
@trmid Once the prize pool is upgraded to be ownable, this will be possible by swapping out the draw manager. This means the variable can stay immutable and be very cheap gas-wise.

@PierrickGT My concern is around liveness, or if there is not enough competition.

- If Optimism has any downtime, the algorithm will jump ahead in time and the reward fraction will be much bigger
- If the Draw bot that Chuck has created is delayed in any way, then the fraction will be much larger

The reserve could grow to hundreds of thousands of POOL. I expect the cap to be extremely high, like 10k POOL, so it's as if there wasn't a cap.

If need be, gov could replace the draw manager with an uncapped one.

**Review by trmid — APPROVED** [review]

File: src/RngRelayAuction.sol [hunk 1]
```diff
@@ -116,6 +121,7 @@ contract RngRelayAuction is IRngAuctionRelayListener, IAuction {
     _auctionTargetTimeFraction = UD2x18.wrap(
       uint64(convert(auctionTargetTime_).div(convert(_auctionDurationSeconds)).unwrap())
     );
+    maxRewards = _maxRewards;
```
[the remark this answers is not in the document]
Comment (asselstine) [reply src/RngRelayAuction.sol:124]: ah yes

**Pull request merged**